### ISSUE TITLE
java.lang.IllegalStateException: Cannot perform backprop: Dropout mask array is absent (already cleared?)

### ISSUE THREAD

**Issue by wangfeng-skymind** [deeplearning4j/deeplearning4j#6756]
Description : this deeplearning4j version  is 1.0.0-beta3
Run some code , and then show some exception, below  code 
is only a little part    ==================================
 ZooModel zooModel = ResNet50.builder().workspaceMode(WorkspaceMode.NONE).build();
        ComputationGraph resnet50 = (ComputationGraph) zooModel.initPretrained();
        System.out.println(resnet50.summary());

        FineTuneConfiguration fineTuneConf = new FineTuneConfiguration.Builder()
                .activation(Activation.LEAKYRELU)
                .weightInit(WeightInit.RELU)
                .updater(new Nesterovs(5e-5))
                .dropOut(0.5)
                .seed(seed)
                .build();

        graph = new TransferLearning.GraphBuilder(resnet50)
                .fineTuneConfiguration(fineTuneConf)
                .setInputTypes(InputType.convolutional(height, width, channels))
                .removeVertexKeepConnections("conv1")
                .addLayer("conv1", new ConvolutionLayer.Builder(new int[]{3, 3})
                        .nIn(channels).nOut(64).activation( Activation.RELU).build(), "input_1")
                .addLayer("fc2048",new DenseLayer.Builder().activation(Activation.TANH).nIn(1000).nOut(2048).build(),"fc1000")
                .removeVertexAndConnections("output")
                .addLayer("newOutput",new OutputLayer
                        .Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                        .activation(Activation.SOFTMAX)
                        .nIn(2048)
                        .nOut(numLabels)
                        .build(),"fc2048")
                .setOutputs("newOutput")
                .build();
        System.out.println(graph.summary());

Exception message=============================================
Epoch=====================0
11:02:48.307 [AMDSI prefetch thread] DEBUG org.nd4j.linalg.memory.abstracts.Nd4jWorkspace - Steps: 4
Exception in thread "main" java.lang.IllegalStateException: Cannot perform backprop: Dropout mask array is absent (already cleared?)
	at org.nd4j.base.Preconditions.throwStateEx(Preconditions.java:641)
	at org.nd4j.base.Preconditions.checkState(Preconditions.java:268)
	at org.deeplearning4j.nn.conf.dropout.Dropout.backprop(Dropout.java:154)
	at org.deeplearning4j.nn.layers.AbstractLayer.backpropDropOutIfPresent(AbstractLayer.java:309)
	at org.deeplearning4j.nn.layers.convolution.subsampling.SubsamplingLayer.backpropGradient(SubsamplingLayer.java:278)
	at org.deeplearning4j.nn.graph.vertex.impl.LayerVertex.doBackward(LayerVertex.java:148)
	at org.deeplearning4j.nn.graph.ComputationGraph.calcBackpropGradients(ComputationGraph.java:2621)
	at org.deeplearning4j.nn.graph.ComputationGraph.computeGradientAndScore(ComputationGraph.java:1369)
	at org.deeplearning4j.nn.graph.ComputationGraph.computeGradientAndScore(ComputationGraph.java:1329)
	at org.deeplearning4j.optimize.solvers.BaseOptimizer.gradientAndScore(BaseOptimizer.java:160)
	at org.deeplearning4j.optimize.solvers.StochasticGradientDescent.optimize(StochasticGradientDescent.java:63)
	at org.deeplearning4j.optimize.Solver.optimize(Solver.java:52)
	at org.deeplearning4j.nn.graph.ComputationGraph.fitHelper(ComputationGraph.java:1149)
	at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:1098)
	at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:1065)
	at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:999)

**Comment by AlexDBlack** [comment]
Thanks for reporting. Fixed here: https://github.com/deeplearning4j/deeplearning4j/pull/6758
Will be merged once tests are complete. A few hours after the merge, fix should be accessible on snapshots: https://deeplearning4j.org/docs/latest/deeplearning4j-config-snapshots

Alternatively, here's a workaround for 1.0.0-beta3:
```
        for(Layer l : graph.getLayers()){
            org.deeplearning4j.nn.conf.layers.Layer conf = (org.deeplearning4j.nn.conf.layers.Layer)l.getConfig();
            if(!(l.getConfig() instanceof org.deeplearning4j.nn.conf.layers.BaseLayer)){
                conf.setIDropout(null);
            } else {
                IDropout d = conf.getIDropout();
                conf.setIDropout(d == null ? null : d.clone());
            }
        }
```
https://gist.github.com/AlexDBlack/339991f8da0a5553223ccaaa690c4507

**Comment by wangfeng-skymind** [comment]
okay,thanks
